lib/ansible/modules/network/checkpoint/cp_gaia_physical_interface.py

```diff
@@ -0,0 +1,158 @@
+    required: false
+    type: str
+  ipv4_mask_length:
+    description: Interface IPv4 address mask length.
+    required: false
+    type: str
+  ipv6_address:
+    description: Interface IPv6 address.
+    required: false
+    type: str
+  ipv6_autoconfig:
+    description: Configure IPv6 auto-configuration.
+    required: false
+    type: bool
+  ipv6_mask_length:
+    description: Interface IPv6 address mask length.
+    required: false
+    type: str
+  mac_addr:
+    description: Configure hardware address.
+    required: false
+    type: str
+  monitor_mode:
+    description: set monitor mode for the interface off/on.
+    required: false
+    type: bool
+  mtu:
+    description: interface mtu.
+    required: false
+    type: int
+  name:
+    description: interface name.
+    required: true
+    type: str
+  rx_ringsize:
+    description: Set receive buffer size for interface.
+    required: false
+    type: int
+  speed:
+    description: Interface link speed. Speed is not relevant when 'auto-negotiation' is enabled
+    required: false
+    type: str
+  tx_ringsize:
+    description: Set transmit buffer size for interfaces.
+    required: false
+    type: int
+short_description: Set Physical interface
+version_added: '2.9'
+
+"""
+
+EXAMPLES = """
+- name: Set comment field of a physical interface
+  cp_gaia_physical_interface:
+    comments: eth0 interface
+    enabled: true
+    name: eth0
+
+"""
+
+RETURN = """
+cp_gaia_physical_interface:
+  description: The updated interface details.
+  returned: always.
+  type: dict
+"""
+
+import ast
+import sys
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.checkpoint.checkpoint import api_call_gaia
+
+
+def main():
+    # arguments for the module:
+    fields = ast.literal_eval("""{"monitor_mode": {"required": False, "type": "bool"}, "ipv6_address": {"required": False, "type": "str"},
+    "ipv4_mask_length": {"required": False, "type": "str"}, "name": {"required": True, "type": "str"}, "duplex": {"required": False, "type": "str"},
+    "tx_ringsize": {"required": False, "type": "int"}, "ipv6_autoconfig": {"required": False, "type": "bool"}, "enabled": {"required": False, "type": "bool"},
+    "comments": {"required": False, "type": "str"}, "mtu": {"required": False, "type": "int"}, "ipv4_address": {"required": False, "type": "str"},
+    "auto_negotiation": {"required": False, "type": "bool"}, "mac_addr": {"required": False, "type": "str"}, "rx_ringsize": {"required": False, "type": "int"},
+    "speed": {"required": False, "type": "str"}, "ipv6_mask_length": {"required": False, "type": "str"}}""")
+    module = AnsibleModule(argument_spec=fields, supports_check_mode=True)
+    was_changed = False
+    ignore = ["status"]
+
+    modules_params_original = module.params
+
+    module_params_show = dict((k, v) for k, v in module.params.items() if k in ["name"] and v is not None)
+
+    module.params = module_params_show
```

Comment:
as I see the implementation here, I am not sure why u are calling the api in sequence of `show->set->show`, Ideally for the respective module only `set-api` should get called and it's respective response should be displayed to the user. So, you'll need to remove the calls to `show-api`.

Reply:
Only `set-api` is needed, but we need the `show-api` in order to achieve idem-potency. when we call just the `set-api` we don't know if something really changed, so we call the `show-api` before and after to check if there were changes following the `set-api`. 
Tell me if you think it's redundant and it's ok to return always the same value (`true`/`false`) on `was_changed`.

Reply:
@chkp-yuvalfe ok, so from `gaia` side something similar to `equals` api that's implemented from mgmt side is not implemented, correct?
As, currently mgmt modules relies on `equals` api to get the content and match with input and check for idempotency